### DISCUSSION TITLE
Strange artifacts in `render_3d` output

### DISCUSSION THREAD

**Discussion by cdwilson** [INTI-CMNB/KiBot#263] [Q&A] [answered]
I'm seeing some strange artifacts in the `render_3d` output.  I'm curious if anybody else has run across this before and was able to diagnose/fix the issue:

![image](https://user-images.githubusercontent.com/14631/186285542-eb2505b9-947e-455d-9516-3ad4c5519396.png)

If I enable `ray_tracing: true` the artifacts go away:

![image](https://user-images.githubusercontent.com/14631/186285577-72f919f1-a0c2-42d6-b1df-041482caa224.png)

I also don't see any artifacts when I open the 3D viewer in KiCad:

<img width="1496" alt="image" src="https://user-images.githubusercontent.com/14631/186285705-98c0bdbe-be47-4137-a34b-ffb010239382.png">

**Comment by set-soft** [comment, marked as answer]
Hi @cdwilson !

Yes, the software render in the virtual server produces these artifacts.
They could be related to the antialiasing, but when I tried disabling it I didn't notice any positive change.
This isn't KiCad or KiBot fault, just the virtual server, and most probably something KiCad does in a particular way.
The solution, as you already know, is to just enable the ray tracer. You get a much better result, is slower, but the hole process is slow anyways.
From my point of view the normal render is just like a draft.
Also note that what you get from the GUI in an interactive session is the result of the hardware accelerated OpenGL, this means the quality depends on the video board and the OpenGL driver.
